This partially reverts commit d003b04f9ff0b2dd71cf8564ba45c164f8e5046c.

The commit introduced in version 2.18.2 not only fixes a side-channel
vulnerability, it also prevents acceptance of malformed inputs - the
ones without the leading zero. This is a breaking change, which breaks
applications depending on the prior behavior, which was introduced in
the fix for CVE-2016-8871 in version 1.11.33, commit
3fb31cef450cef82015170f8e825a2d656163ea6.

Based on CT::strip_leading_zeros().

--- b/src/lib/pk_pad/eme_oaep/oaep.cpp
+++ a/src/lib/pk_pad/eme_oaep/oaep.cpp
@@ -72,9 +72,9 @@
    Therefore, the first byte can always be skipped safely.
    */
 
-   const auto leading_0 = CT::Mask<uint8_t>::is_zero(in[0]);
+   const uint8_t skip_first = CT::Mask<uint8_t>::is_zero(in[0]).if_set_return(1);
 
-   secure_vector<uint8_t> input(in + 1, in + in_length);
+   secure_vector<uint8_t> input = CT::copy_output(CT::Mask<uint8_t>::cleared(), in, in_length, skip_first);
 
    const size_t hlen = m_Phash.size();
 
@@ -86,9 +86,7 @@
              input.data(), hlen,
              &input[hlen], input.size() - hlen);
 
-   auto unpadded = oaep_find_delim(valid_mask, input.data(), input.size(), m_Phash);
-   valid_mask &= leading_0.unpoisoned_value();
-   return unpadded;
+   return oaep_find_delim(valid_mask, input.data(), input.size(), m_Phash);
    }
 
 secure_vector<uint8_t>
